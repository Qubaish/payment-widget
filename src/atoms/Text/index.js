import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SmallText = styled.p`
    font-size: 14px;
    color: #d8d8d8;
    margin: 0;
`;
const LargeText = styled.p`
    font-size: 16px;
    margin: 0;
`;

function Text({text, size}){
    return (
        <Fragment>
            {size === 'sm' && <SmallText data-testid="text">{text}</SmallText>}
            {size === 'lg' && <LargeText data-testid="text">{text}</LargeText>}
        </Fragment>
    );
}

Text.propTypes = {
    text: PropTypes.string,
    size: PropTypes.string,
};

export default Text;