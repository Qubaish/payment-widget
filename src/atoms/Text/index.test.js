import React from 'react';
import { render } from '@testing-library/react';
import Text from './index';

describe('<Text />', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<Text text="hello"/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('should render a text', () => {
        const { getByTestId} = render(<Text text="test" size="sm"/>);
        const elem = getByTestId('text');
        expect(elem.innerHTML).toBe('test');
    });
    it('should render a large text', () => {
        const { getByTestId} = render(<Text text="test" size="lg"/>);
        const elem = getByTestId('text');
        expect(elem).toHaveStyle('font-size: 16px');
    });
});