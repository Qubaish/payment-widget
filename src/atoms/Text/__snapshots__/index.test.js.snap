// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Message /> checking snapshot 1`] = `<DocumentFragment />`;

exports[`<Text /> checking snapshot 1`] = `<DocumentFragment />`;
